### PR TITLE
ceph-perf-pull-requests: use clang for crimson perf test

### PR DESCRIPTION
**Pull request by tchaikov**

### CODE REVIEW

**Comment by tchaikov** [comment]
@AmnonHanuhov

File: ceph-perf-pull-requests/config/definitions/ceph-perf-pull-requests.yml [hunk 1]
```diff
@@ -56,9 +56,21 @@
           fi
           export NPROC=$(nproc)
           export FOR_MAKE_CHECK=true
+          cxx_compiler=g++
+          c_compiler=gcc
+          for i in $(seq 11 -1 8); do
```
Comment (rzarzynski) [inline ceph-perf-pull-requests/config/definitions/ceph-perf-pull-requests.yml:61]: We're doing auto-detection of available Clang's version with a fallback to GCC.

File: ceph-perf-pull-requests/config/definitions/ceph-perf-pull-requests.yml [hunk 2]
```diff
@@ -56,9 +56,21 @@
...
+                  c_compiler="clang-$i"
+                  break
+              fi
+          done
           if test {osd-flavor} = "crimson" ; then
               export WITH_SEASTAR=true
-              timeout 7200 src/script/run-make.sh --cmake-args "-DCMAKE_BUILD_TYPE=Release -DWITH_SEASTAR=ON -DWITH_TESTS=OFF" vstart-base crimson-osd
+              # TODO use clang-10 on ubuntu/focal
+              timeout 7200 src/script/run-make.sh \
+              --cmake-args "-DCMAKE_CXX_COMPILER=$cxx_compiler -DCMAKE_C_COMPILER=$c_compiler -DCMAKE_BUILD_TYPE=Release -DWITH_SEASTAR=ON -DWITH_TESTS=OFF" \
```
Comment (rzarzynski) [inline ceph-perf-pull-requests/config/definitions/ceph-perf-pull-requests.yml:72]: nit: extra indent?

File: ceph-perf-pull-requests/config/definitions/ceph-perf-pull-requests.yml [hunk 3]
```diff
@@ -56,9 +56,21 @@
...
+              vstart-base crimson-osd
               src/script/run-cbt.sh --build-dir $PWD/build --source-dir $PWD --cbt ${{WORKSPACE}}/cbt -a $archive_dir src/test/crimson/cbt/radosbench_4K_read.yaml
           else
```
Comment (rzarzynski) [inline ceph-perf-pull-requests/config/definitions/ceph-perf-pull-requests.yml:75]: ACK, classical stays unaffected.

**Comment by tchaikov** [comment]
@rzarzynski we need have https://github.com/ceph/ceph/pull/35336 first.

File: ceph-perf-pull-requests/config/definitions/ceph-perf-pull-requests.yml (hunk 1 above)
Comment (tchaikov) [reply ceph-perf-pull-requests/config/definitions/ceph-perf-pull-requests.yml:61]: yes, starts from 11 for future-proof.

**Comment by ronen-fr** [comment]
@mkogan1 FYI (and see the ref'ed)

**Pull request merged**